src/os/ui/featureedit.js

```diff
@@ -1154,9 +1196,21 @@ os.ui.FeatureEditCtrl.prototype.saveGeometry_ = function(feature) {
     // make sure the coordinate values are numeric
     var lon = Number(this['pointGeometry']['lon']);
     var lat = Number(this['pointGeometry']['lat']);
+    var alt = Number(this['altitude']) ? this['altitude'] : null;
```

Comment:
You can assume 0 for null/undefined altitude here as well, so `Number(this['altitude']) || 0`.

For the units, use values from `os.math.Units` instead of keys, changing the statement to `var altU = this['altUnits'] || os.math.Units.METERS;`.

Given those two changes, you can always set altitude on the coordinate, do the unit conversion, etc, instead of checking if both values are defined first.